tutorials/marble/version-control.md

```diff
@@ -0,0 +1,393 @@
+
+![Git Staged](images/version-control/git-staged.png)
+
+### Making a Commit
+
+Select the file and fill in the Title/Summary for the commit.  It should be short and descriptive.  
+It can be as simple as "Updated Untitled.ipynb" or more specific, such as "Added visualization for rainfall 2022".
+
+A description is recommended but is not required.
+
+![Git Commit Title](images/version-control/git-commit-title.png)
+
+When done click the `Commit` button below the commit description.  If successful you will see a `Committed changes` status pop up in the lower right corner.
+
+![Git Commit Successful](images/version-control/git-commit-successful.png)
+
+### Pushing a Commit
+
+Making a commit sets a "flag" in the progress of your work that shows what has been done at certain points. Pushing a commit moves these changes to the remote 
+repository where they can be merged with the master/main branch.  Up to now no branches have been made.
+
+Click the `Git` menu and `Push to Remote`
+
+![Git Push Menu](images/version-control/git-push-menu.png)
+
+Enter in your login credentials.  For the password enter your Personal Access Token.  Click `OK`.
+
+
+![Git Push Credentials](images/version-control/git-push-credentials.png)
+
+If successful you will see a `Successfully pushed` status pop up in the lower right corner.
+
+![Push Successful](images/version-control/push-successful.png)
+
+### Creating a Branch
+
+Click the down arrow in the `Current Branch` section in the Git sidebar.  It will reveal the `Branches` window and show all the created branches.
+
+Click the `New Branch` button.
+
+![New Branch Button](images/version-control/new-branch-button.png)
+
+Fill out the form and input a branch name, and click which branch to base the new branch off of.  After that click `Create Branch`.
+
+It is good practice to not use spaces or special characters in the branch name other than dashes.
+
+
+![New Branch Form](images/version-control/new-branch-dialog.png)
+
+Git will automatically switch you into the new branch.  Your newly created branch will be shown under `Current Branch` and be in the list of branches.  
+
+![Create Branch Successful](images/version-control/create-branch-successful.png)
+
+### Stashing Files
+
+Stashing a file means to temporarily store files you're working on so the changes made to them will not affect other files, 
+and changes to other files will not affect the working file.  Stashing is typically done before switching branches, so 
+the changes from other branches will not overwrite changes in files you are currently working on.
+
+Currently there is no mechanism in the GUI to stash files.  Files can only be stashed from the Terminal.
+
+[Stash Files from Terminal](#stash-terminal)
+
+Once the command is run all files listed under `Changed` will be moved to the `Stash` and will not be seen. 
+
+
+### Switching Branches
+
+```{note}
+You can only switch branches when you have no files staged or changed.
```

Comment:
You can switch branches with changed files as long as the change does not result in a conflict

Reply:
Changed text to include conflicting files